arch/arm/Kconfig

```diff
@@ -53,6 +53,12 @@ config ARM_ON_ENTER_CPU_IDLE_HOOK
 	  If needed, this hook can be used to prevent the CPU from actually
 	  entering sleep by skipping the WFE/WFI instruction.
 
+config ARM_ON_EXIT_CPU_IDLE_HOOK
```

Comment:
The problem is that this is mimicking `CONFIG_ARM_ON_ENTER_CPU_IDLE_HOOK` that is indeed a real hook, this is not, this is an injected macro basically.

So, we have two options here:
1. We keep mirroring what is done for `CONFIG_ARM_ON_ENTER_CPU_IDLE_HOOK` that is we add a real `z_arm_on_exit_cpu_idle` function that is called with all the the needed pre/post-amble of `push / pop` and we keep this name.
2. We use the current solution but we change symbol name and we justify it why this is not an hook but an instructions injection by macro (to me, that makes much more sense because you are saving at least a couple of instructions but on the downside you must write your routine in assembly, while with the hook you can write it in C)

Reply:
I will pick 2nd option. Calling hook written in C is not an option because it might push sth on the stack, memory access fust after wfi is this thigh we need to avoid.

Reply:
yes, this is exactly the kind of justification that is needed upstream

Reply:
fixed, picked 2nd option